src/python/pants/jvm/jdk_rules.py

```diff
@@ -355,12 +379,37 @@ async def jvm_process(bash: BashBinary, request: JvmProcess) -> Process:
         **request.extra_env,
     }
 
+    def valid_jvm_opt(opt: str) -> str:
+        if opt.startswith("-Xmx"):
+            raise ValueError(
+                softwrap(
+                    f"""
+                    Invalid value for JVM options: {opt}.
+
+                    For setting a maximum heap size for the JVM child processes, use
+                    `[GLOBAL].child_process_default_memory_usage` option instead.
+
+                    Run `{bin_name()} help-advanced global` for more information.
+                    """
```

Comment:
🤦 

This is a _way_ better idea than doing the templating, heh. Thanks!

Reply:
yeah, came up with this because the templating implementation was very naive and, doing it in the nailgun command runner, meant not being able to pass JVM options to processes where `use_nailgun=False` was set, and I thought there was no good reason to not allow passing, for example, system prooperties to those non-nailgun processes.